llvm/lib/Target/AMDGPU/AMDGPUCombinerHelper.cpp

```diff
@@ -445,3 +445,75 @@ void AMDGPUCombinerHelper::applyExpandPromotedF16FMed3(MachineInstr &MI,
   Builder.buildFMinNumIEEE(MI.getOperand(0), B1, C1);
   MI.eraseFromParent();
 }
+
+bool AMDGPUCombinerHelper::matchCombineFmulWithSelectToLdexp(
+    MachineInstr &MI, MachineInstr &Sel,
+    std::function<void(MachineIRBuilder &)> &MatchInfo) {
+  assert(MI.getOpcode() == TargetOpcode::G_FMUL);
+  assert(Sel.getOpcode() == TargetOpcode::G_SELECT);
+
+  Register Dst = MI.getOperand(0).getReg();
+  LLT DestTy = MRI.getType(Dst);
+  LLT ScalarDestTy = DestTy.getScalarType();
+
+  if ((ScalarDestTy == LLT::float64() || ScalarDestTy == LLT::float32() ||
```

Comment:
Did we also handle bfloat in the dag?

Reply:
Nope, anyways in globalIsel LLT accounts just for collection of bits (implying both LLT::float16 and LLT::int16 and LLT::bfloat16, are representing just LLT::scalar(16)). So, does it matter?

Reply:
Yes, it does matter. bfloat is getting added

Reply:
I thought you added bfloat support and bfloat tests in the original DAG patch? I don't see them